dev/src/resources/server.rs

```diff
@@ -50,6 +53,72 @@ impl Server {
         Ok(server)
     }
 
+    pub async fn serve_web_sockets() -> Result<Self> {
+        let ip = random_loopback_ip();
+        let address = SocketAddr::new(ip.into(), 8080);
+
+        let tcp_listener = TcpListener::bind(address).await.expect("failed to bind");
+
+        let server_handle = tokio::spawn(async move {
+            while let Ok((stream, _)) = tcp_listener.accept().await {
+                let ws_stream = tokio_tungstenite::accept_async(stream)
+                    .await
+                    .expect("Error during the websocket handshake occurred");
+                let (write, read) = ws_stream.split();
+                read.try_filter(|msg| future::ready(msg.is_text() || msg.is_binary()))
+                    .forward(write)
+                    .await
+                    .expect("Failed to forward messages");
+            }
+        });
+
+        let server = Server {
+            server_handle,
+            address,
+        };
+
+        server.wait_ready_socket().await?;
```

Comment:
For a regular HTTPS server, we use `wait_ready` to wait until the server is listening. It looks like `wait_ready_socket` instead “tests the test” by sending some messages and getting them back. I don't think testing the test is necessary here, because the test code already tests the same behavior.

As for waiting for the server to become ready, ideally, we should write `wait_ready` so that it works for both regular and websocket servers; I'm not a fan of having to call different methods depending on how a struct was constructed. But because this is just test code, I'm fine with the approach of removing `wait_ready_socket` entirely, unless it has importance that I haven't covered? It seems like we don't need to explicitly wait for the server to become ready for the tests to pass.